### PR TITLE
Return API Users with distinct result when using Ransack

### LINKED ISSUE #3183: User picker proposes the same users at promotion edit page
When we have a user, who has few addresses then he will be proposed with user_picker.js multiple times.

**Solidus Version**:
2.4.2 (checked at https://solidus-master-demo.herokuapp.com/ also)
[Gemfiles](https://gist.github.com/VitaliyAdamkov/738512cfe887ead87d852be34c5ec99a)

**To Reproduce**
1. Make an order for default user with Address1.
2. Make an order for default user with Address2.
3. Create a promotion.
4. Go to the promotion edit page.
5. Add the user promotion rule.
6. Type any letter from default user's email.

**Current behavior**
API call runs and select shows default user at least two times.

**Expected behavior**
Select is expected to show unique users.

**Screenshots**
изображение

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Browser Mozilla Firefox
 - Version 66.0.2

### PR DESCRIPTION
**Pull request by hefan**
**Description**
Return API Users ransack searches with a distinct result.

Fixes #3183
-->

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [ ] I have added tests to cover this change (if needed)
- [ ] I have attached screenshots to this PR for visual changes (if needed)

### CODE REVIEW

**Comment by hefan** [comment]
I was not quite sure to write one.
The Request Spec for spree/api/users_controller has no scenario for that.
Should i write one which tests that each user is only once returned when using a ransack search like the one in #3183?
Can you point me to where there is a similar test like this so i could follow the best practices from the start? And where the test should go to (i guess only to requests/spree/api/users_controller_spec.rb)?

**Comment by jarednorman** [comment]
Yeah, I think a request spec in the file `requests/spree/api/users_controller_spec.rb` would be appropriate. Even just a single spec that fails without this change and passes with it, showing that the endpoint returns non-duplicated users would be great.

**Comment by spaghetticode** [comment]
I was wondering if having distinct results may be something we may want to move upstream, ie. avoid that any subclass of Spree::Api::ResourceController returns duplicates, not only this one.

**Comment by kennyadsl** [comment]
@spaghetticode this was the approach used in [another PR](https://github.com/solidusio/solidus/pull/3195) for the same issue. Please see this comment https://github.com/solidusio/solidus/pull/3195#issuecomment-500366500 and let me know any thoughts on that.

**Comment by hefan** [comment]
I think about the considerations from @kennyadsl in #3195:

1. i do not understand why not and how does `distinct: true` as ransack option does the same like the activerecord distinct. But i assume the activerecord distinct is not a problem from what i read there.
2. The special handling in the backend users controller is gone in master.

However, about the use in the resource controller i am not sure. i imagine scenarios where one could want none distinct results when searching for product/variants or orders/line items etc.

My first idea was to just remove 
line 31: `firstname_or_lastname_start: term` here
https://github.com/solidusio/solidus/blob/16ed773a6554782e33476e2ffc41b70320ddb46e/backend/app/assets/javascripts/spree/backend/user_picker.js#L31

because i never would search for an user by an adress when i want to give promotions on a user basis. Its the only place where the user picker is used.

Please tell me about the preferred way to go here..

**Comment by hefan** [comment]
OK. i've added a spec test for the original approach of this PR with a distinct API Users ransack search result for users

File: api/spec/requests/spree/api/users_controller_spec.rb [hunk 1]
```diff
@@ -144,6 +144,21 @@ module Spree
         expect(response.status).to eq(422)
         expect(json_response).to eq({ "error" => "Cannot delete record." })
       end
+
+      it "returns distinct search results" do
+        uuser = create(:user, email: 'distincttest@solidus.com')
```
Comment (aldesantis) [inline api/spec/requests/spree/api/users_controller_spec.rb:149]: Small typo here, should be `user`?

Comment (hefan) [reply api/spec/requests/spree/api/users_controller_spec.rb]: It was meant to be an abbreviation of unique_user. i have changed the spec it to be more precise.

Comment (aldesantis) [reply api/spec/requests/spree/api/users_controller_spec.rb]: Ah, got it! Yeah, I think the full version is more easily understandable.

**Comment by spaghetticode** [comment]
@hefan can you please rebase with master? This build is failing due to an annoying flaky spec that was fixed recently. After the rebase your build should become green 💚 ... thanks 🙏

**Comment by hefan** [comment]
done

**Comment by kennyadsl** [comment]
Thank you Stefan!

**Pull request merged**